expo/expo
# Expo font/allow weights on android

# Why

I'm working on a expo (web/ios/android) project using tailwind. Having font weights on all platforms improves the productivity of our team since the web team is used using tailwind by providing font weights instead of changing the font family.

# How
I made an backwards compatible change in the font plugin api. It gives the option to explicitly define weight, family and text style. From these values an xml is generated and referenced in the mainApplication. 

It basically generates the code explained [here](https://github.com/jsamr/react-native-font-demo).

# Test Plan
1. Follow the docs
2. Run expo prebuild
3. Check if correct xml is generated in res/font and is referenced mainApplication
4. Run iOS and Android to check if weights are working.


# Checklist
